src/sql/workbench/services/connection/browser/connectionManagementService.ts

```diff
@@ -925,56 +926,76 @@ export class ConnectionManagementService extends Disposable implements IConnecti
 
 	/**
 	 * Refresh Azure access token if it's expired.
-	 * @param uri connection uri
+	 * @param uriOrConnectionProfile connection uri or connection profile
 	 * @returns true if no need to refresh or successfully refreshed token
 	 */
-	public async refreshAzureAccountTokenIfNecessary(uri: string): Promise<boolean> {
-		const profile = this._connectionStatusManager.getConnectionProfile(uri);
-		if (!profile) {
-			this._logService.warn(`Connection not found for uri ${uri}`);
+	public async refreshAzureAccountTokenIfNecessary(uriOrConnectionProfile: string | ConnectionProfile): Promise<boolean> {
+		if (!uriOrConnectionProfile) {
```

Comment:
If undefined is actually expected then you should update the parameter to `string | ConnectionProfile | undefined`
```suggestion
	public async refreshAzureAccountTokenIfNecessary(uriOrConnectionProfile: string | ConnectionProfile | undefined): Promise<boolean> {
		if (!uriOrConnectionProfile) {
```

Reply:
It's not expected I would say, this check is to handle any possible NRE.